api/pkg/controller/rbac-user-cluster/machinedeployment/controller.go

```
@@ -0,0 +1,80 @@
+package machinedeployment
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	apiextensionv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const machinedeploymentCRDName = "machinedeployments.cluster.k8s.io"
+
+// Add creates a new Namespace Controller and adds it to the Manager.
+// The Manager will set fields on the Controller and Start it.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		// If we can't get a clientset we can't do anything else
+		glog.Fatal(err)
```

please return the error instead.

---

done